src/events/hooks/useEventsRepoState.ts

```diff
@@ -1,21 +1,57 @@
 import { DateTime } from 'luxon';
-import { useEffect, useState } from 'react';
+import { useContext, useEffect, useMemo, useState } from 'react';
 
 import { getCalendarEvents } from 'events/api/calendarEvents';
-import { getEvent, getEvents, IEventAPIParameters } from 'events/api/events';
+import { getAllEventsParamtered, getEvent, getEvents, IEventAPIParameters } from 'events/api/events';
 import { INewEvent } from 'events/models/Event';
+import { useQueryParamsState } from '../../core/hooks/useQueryParamsState';
+import { QueryParams } from '../../core/providers/QueryParams';
+import { isAfter, isBefore, isInDateRange } from '../utils/eventTimeUtils';
 
 export type EventMap = Map<number, INewEvent>;
 
 const INITIAL_STATE: INewEvent[] = [];
 
+const getFilteredEventList = (searchContext: ReturnType<typeof useQueryParamsState>, eventList: INewEvent[]) => {
```

Comment:
This hook/function is becoming quite large.
Should probably try to split some of the logic is separate functions instead of inside the function body.

Should also try using `useMemo` and `useCallback` in apropriate places to mitiagte performance issues.
Remember that the _entire_ hook and all subscribed children of the context will be read and re-rendered when _anything_ in the state is changed.